src/SoundFingerprinting/Query/Extensions.cs

```diff
@@ -20,14 +20,14 @@ public static double StdDev(this IEnumerable<double> values)
             return ret;
         }
 
-        public static IEnumerable<Discontinuity> FindGaps(this IEnumerable<Tuple<uint, float>> entries, double permittedGap)
+        public static IEnumerable<Discontinuity> FindGaps(this IEnumerable<Tuple<uint, float>> entries, double permittedGap, double fingerprintLength)
         {
             var floats = entries.OrderBy(entry => entry.Item2).ToArray();
             for (int i = 1; i < floats.Length; ++i)
             {
-                if (floats[i].Item2 - floats[i - 1].Item2 > permittedGap && floats[i].Item1 - floats[i - 1].Item1 > 1)
+                if (floats[i].Item2 - (floats[i - 1].Item2 + fingerprintLength) > permittedGap && floats[i].Item1 - floats[i - 1].Item1 > 1)
```

Comment:
Use the SubFingerprintsToSeconds class here to calculate the exact length of the gap.

Reply:
Ok.